gptel-context.el

```diff
@@ -163,32 +164,62 @@ context chunk.  This is accessible as, for example:
         (eq buffer-file-coding-system 'no-conversion))
     (file-missing (message "File \"%s\" is not readable." path))))
 
+(defun gptel-context--handle-binary (path)
+  "Add binary file at PATH to context if supported.
+Return PATH if added, nil if ignored."
+  (if-let* (((gptel--model-capable-p 'media))
+            (mime (mailcap-file-name-to-mime-type path))
+            ((gptel--model-mime-capable-p mime)))
+      (prog1 path
+        (cl-pushnew (list path :mime mime)
+                    gptel-context--alist :test #'equal)
+        (message "File \"%s\" added to context." path))
+    (message "Ignoring unsupported binary file \"%s\"." path)
+    nil))
+
+(defun gptel-context--handle-directory (path action)
+  "Process all files in directory at PATH according to ACTION.
+ACTION should be either `add' or `remove'."
+  (cl-block gptel-context--handle-directory
+    (when (eq action 'add)
+      (unless (y-or-n-p (format "Recursively add all files from directory %s? " path))
```

Comment:
`gptel-context--handle-directory` is a non-interactive function.  Running an interactive `y-or-n-p` check inside a non-interactive function is not a good idea, since `gptel-add` can be used non-interactively for scripting.  [Here's an example](https://github.com/meain/dotfiles/blob/master/emacs/.config/emacs/init.el#L4525).  So this check should happen in `gptel-context-add-file`.

Reply:
> The other advantage is that we only need to ask once per call to `gptel-add`, which is less annoying than asking per-directory.

Apologies if I’m misunderstanding something, but since `gptel-add` calls `gptel-context-add-file` once per directory, moving the logic to the latter would not by default have the effect you describe, I think. Can you clarify?

Reply:
@benthamite Right on cue! https://github.com/karthink/gptel/issues/513#issuecomment-2560155884

Reply:
> You are not adding all files inside a directory recursively -- instead you're only descending one level

I don’t think this is correct: the current functionality adds all files inside a directory, regardless of whether these files are just one level up or in other subdirectories. If you think I’m mistaken, could you provide an example where the function doesn’t do the above?

Reply:
Separately, I have moved the `y-or-n-p` prompt from `gptel-context-add` to `gptel-context-add`. This also prompts once per call rather than once per directory.

Reply:
> I don’t think this is correct: the current functionality adds all files inside a directory

Sorry, you're right -- I confused myself twice there.